iOS/Objective C/IronSourceDemoApp/NativeAdView.m

```diff
@@ -0,0 +1,177 @@
+        [_topView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor constant:8],
+        [_topView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor constant:-8],
+        [_topView.heightAnchor constraintEqualToConstant:30],
+        
+        [_adBadge.widthAnchor constraintEqualToConstant:20],
+        [_adBadge.heightAnchor constraintEqualToConstant:20],
+        [_adBadge.leadingAnchor constraintEqualToAnchor:_topView.leadingAnchor],
+        [_deleteButton.widthAnchor constraintEqualToConstant:30],
+        [_deleteButton.heightAnchor constraintEqualToAnchor:_topView.heightAnchor],
+        [_deleteButton.trailingAnchor constraintEqualToAnchor:_topView.trailingAnchor],
+        
+        // Set the _topView as the top-most view for the native ad.
+        [self.topAnchor constraintEqualToAnchor:_topView.topAnchor constant:-8],
+        [firstSubview.topAnchor constraintEqualToAnchor:_topView.bottomAnchor constant:8],
+    ]];
+}
+
+- (void)setClearBackgroundRecursively:(UIView *)parentView {
+    if (![parentView isKindOfClass:[UIButton class]]) {
+        parentView.backgroundColor = [UIColor clearColor];
+    }
+    for (UIView *subview in parentView.subviews) {
+        [self setClearBackgroundRecursively:subview];
+    }
+}
+
+// Predefine the colors and constraints of the native ad's properties.
+// Replace with your desired colors and look.
+- (void)setNativeAdViewConstraints {
+    [self setClearBackgroundRecursively:self];
+    self.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    self.adTitleView.textColor = UIColor.blackColor;
+    self.adAdvertiserView.textColor = UIColor.blackColor;
+    self.adBodyView.textColor = UIColor.blackColor;
+    [self.adCallToActionView setTitleColor:[UIColor blueColor] forState:UIControlStateNormal];
+    
+    UIView *mediaView = self.adMediaView;
+    if (mediaView) {
+        for (UIView *subview in mediaView.subviews) {
+            [subview.centerXAnchor constraintEqualToAnchor:self.centerXAnchor].active = YES;
+        }
+    }
+}
+
+// This was triggered after a successful load of a native ad.
+- (void)loadNativeAdLayout:(LevelPlayNativeAd *)nativeAd {
+    _nativeAd = nativeAd;
+    
+    if (_nativeAd.icon.image) {
+        self.adAppIcon.image = _nativeAd.icon.image;
+    } else {
+        [self.adAppIcon removeFromSuperview];
+    }
+    self.adTitleView.text = _nativeAd.title;
+    self.adAdvertiserView.text = _nativeAd.advertiser;
+    self.adBodyView.text = _nativeAd.body;
+    [self.adCallToActionView setTitle:_nativeAd.callToAction forState:UIControlStateNormal];
+    self.adCallToActionView.userInteractionEnabled = NO;
+    
+    // The below part is copied from ISNativeAdView's (void)setNativeAd method.
+    // For some unknow reason these parts were not triggered.
+    // adMediaView and networkNativeAdView was affected the most.
+    // Should have the below single line instead:
+    // [self setNativeAd:nativeAd];
```

Comment:
Could you verify this part please? @maozel

Reply:
solved, please review
Objective-C automatically generates a setter functional called setPropertyName. So for setNativeAd, instead of calling the function defined in ISNativeAd, it called the setter. I changed the  name of _nativeAd to _levelplayNativeAd and it fixed

Reply:
we decided to change the method name to registerNativeAdViews, i will sen you a new xcframework with the change

Reply:
changed, please review